site/app/models/gradeable/Gradeable.php

```diff
@@ -556,6 +599,11 @@ public function setComponents(array $components) {
             }
         }
         $this->components = $components;
+
+        // sort by order
+        usort($this->components, function(Component $a, Component $b) {
+            return $a->getOrder() - $b->getOrder();
+        });
     }
```

Comment:
`setComponents` should update `modified` (I assume)

Reply:
Since changing components doesn't require a write to the `gradeable` or `electronic_gradeable` table, it doesn't.  The components individually get checked for their own `modified` field.

Reply:
Understandable.